.github/workflows/publish.yaml

```
@@ -0,0 +1,31 @@
+name: publish
+on:
+  push:
+    # Sequence of patterns matched against refs/heads
+    branches:    
+      - 'main'
+jobs:
+  build_and_publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+#          token: ${{ secrets.CI_TOKEN }}
+#          submodules: true
+
+#      - name: git submodule
+#        run: git submodule update --init
+
+      -
+        uses: docker/login-action@v1
+        name: Login to GitHub Container Registry
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      -
+        run: |
```

Again with consistent formatting
```suggestion

      - run: |
```

---

Changed!